### ISSUE TITLE
wait_for_successful_query hangs

### ISSUE THREAD

**Issue by dverbeek84** [saltstack/salt#45497]
### Description of Issue/Question
When i use this state:

```
test_github:
  http.wait_for_successful_query:
  - name: 'https://github.com'
  - status: 200
  - request_interval: 10
```

it hangs on the minion.

I get the following debug log:

```
[DEBUG   ] LazyLoaded http.wait_for_successful_query
[INFO    ] Running state [https://github.com] at time 14:04:25.048492
[INFO    ] Executing state http.wait_for_successful_query for [https://github.com]
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
[DEBUG   ] delaying query for 10 seconds.
```

`curl --head https://github.com` output:
```
HTTP/1.1 200 OK
Server: GitHub.com
Date: Wed, 17 Jan 2018 14:11:59 GMT
Content-Type: text/html; charset=utf-8
Status: 200 OK
Cache-Control: no-cache
Vary: X-PJAX
```

`http.query` works fine.

### Versions Report
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.5.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core

**Comment by Ch3LL** [comment]
I'm not able to replicate this issue but glancing at the code its getting to that debug line because this is never true: https://github.com/saltstack/salt/blob/v2017.7.2/salt/states/http.py#L150 which seems odd and it doesn't look like you are setting a custom wait time. 

Can you add this logging information so we can see why that is not true:

```
diff --git a/salt/states/http.py b/salt/states/http.py
index 075ae194d7..1a7b1af018 100644
--- a/salt/states/http.py
+++ b/salt/states/http.py
@@ -147,6 +147,9 @@ def wait_for_successful_query(name, wait_for=300, **kwargs):
         except Exception as exc:
             caught_exception = exc
 
+        log.error(wait_for)
+        log.error(time.time())
+        log.error(starttime + wait_for)
         if time.time() > starttime + wait_for:
             if not ret and caught_exception:
                 # workaround pylint bug https://www.logilab.org/ticket/3207

```

**Comment by gtmanfred** [comment]
This will be fixed in 2017.7.3 https://github.com/saltstack/salt/pull/44968

**Comment by dverbeek84** [comment]
I also don't have the issue after fresh installs of the machine, so i can't reproduce it.
